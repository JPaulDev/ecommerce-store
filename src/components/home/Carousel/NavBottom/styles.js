import styled from 'styled-components';

export const List = styled.ul`
  align-items: center;
  bottom: 15px;
  display: flex;
  left: 50%;
  list-style: none;
  position: absolute;
  transform: translateX(-50%);
  user-select: none;
  z-index: 10;

  li {
    display: flex;
  }

  @media screen and (min-width: 715px) {
    display: none;
  }
`;

export const Button = styled.button`
  background: ${({ isActive }) =>
    isActive ? 'var(--color-orange-1)' : 'white'};
  border-radius: 2px;
  cursor: pointer;
  height: ${({ isActive }) => (isActive ? '7px' : '5px')};
  margin: 0 5px;
  transition: all 100ms ease-in-out;
  width: 33px;
`;
