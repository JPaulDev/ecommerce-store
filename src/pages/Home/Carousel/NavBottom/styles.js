import styled from 'styled-components';

export const Container = styled.div`
  bottom: 10px;
  left: 50%;
  position: absolute;
  transform: translateX(-50%);
  user-select: none;
  z-index: 10;
`;

export const List = styled.ul`
  display: flex;
  list-style: none;

  li {
    align-items: center;
    display: flex;
  }
`;

export const Button = styled.button`
  background: ${({ isActive }) =>
    isActive ? 'var(--color-orange-1)' : 'white'};
  border-radius: 2px;
  cursor: pointer;
  height: ${({ isActive }) => (isActive ? '7px' : '5px')};
  margin: 0 5px;
  transition: all 100ms ease-in-out;
  width: 33px;
`;
