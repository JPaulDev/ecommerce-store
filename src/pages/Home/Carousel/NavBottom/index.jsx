import * as Styled from './styles';

export default function NavBottom({ length, slideIndex, onChangeSlide }) {
  return (
    <Styled.Container>
      <Styled.List>
        {Array.from({ length }).map((_, index) => (
          <li key={index}>
            <Styled.Button
              type="button"
              isActive={index === slideIndex}
              aria-label={`Slide ${index + 1} of ${length}`}
              onClick={() => onChangeSlide('JUMP', index)}
            />
          </li>
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
