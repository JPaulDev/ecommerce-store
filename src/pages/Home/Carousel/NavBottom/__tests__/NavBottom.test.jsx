import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NavBottom from '../index';

describe('NavBottom component', () => {
  it('renders a button for each slide in the slides array', () => {
    const tree = renderer.create(<NavBottom length={2} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies styles to the active slide button', () => {
    render(<NavBottom length={2} slideIndex={0} />);
    const button = screen.getByRole('button', {
      name: /slide 1 of 2/i,
    });

    expect(button).toHaveStyle('height: 7px');
  });
});
