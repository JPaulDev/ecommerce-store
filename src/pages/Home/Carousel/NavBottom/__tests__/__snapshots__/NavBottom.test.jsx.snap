// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBottom component renders a button for each slide in the slides array 1`] = `
<div
  className="sc-bczRLJ daBuPX"
  data-testid="nav-bottom"
>
  <ul
    className="sc-gsnTZi emOVAI"
  >
    <li>
      <button
        aria-label="Slide 1 of 2"
        className="sc-dkzDqf klJSdk"
        onClick={[Function]}
        type="button"
      />
    </li>
    <li>
      <button
        aria-label="Slide 2 of 2"
        className="sc-dkzDqf klJSdk"
        onClick={[Function]}
        type="button"
      />
    </li>
  </ul>
</div>
`;
